go.mod

```diff
@@ -5,6 +5,7 @@ go 1.22.7
 toolchain go1.23.4
 
 require (
+	github.com/ArthurSens/otlp-prometheus-translator v0.0.0-20250228115610-6db8fa672349
```

Comment:
Thanks for splitting this into new module! Would be super useful for downstream callers that support [otlp ingestion like Thanos](https://github.com/thanos-io/thanos/tree/main/pkg/receive/otlptranslator).

I wonder if a good place for this module might be in https://github.com/prometheus/client_golang/tree/main/exp? As this is zero dep already, will probably be treated as experimental for some time and will have several downstream callers. Wdyt? cc: @bwplotka

Reply:
I think it sounds very out of scope for client golang🤔, but super happy to move elsewhere in the Prometheus org

Reply:
Sorry, I didn't fully clarify my thinking there. client_golang scope to me is "instrumentation lib + clients/utilities to interact with Prometheus and its APIs". otlp (like remote write) is an accepted way of interacting with Prometheus, so code for otlp translation, can probably live there as a submodule (near some of the remote write API handling code that is in `exp`). 🙂 

However, not blocking anything, prometheus/common would be good too.

Reply:
Yea, I would recommend separate repo for now, so it can have more external maintainers e.g. in prom org, no harm in that vs it might overload client_golang.